### PR TITLE
Fix salary budget

### PR DESCRIPTION
**Pull request by tobhil98**
# Summary
- Salary and budget for IIGO

## Test Plan

- todo

### CODE REVIEW

File: internal/server/iigointernal/executive.go [hunk 1]
```diff
@@ -105,58 +128,64 @@ func (e *executive) replyAllocationRequest(commonPool shared.Resources) {
...
+	return e.clientPresident.SetTaxationAmount(islandsResources)
 }
 
 func getIslandAlive() []float64 {
 	return rules.VariableMap[rules.IslandsAlive].Values
 }
+
+// incur charges in both budget and commonpool for performing an actions
+// actionID is typically the function name of the action
+// only return error if we can't withdraw from common pool
```
Comment (MikeIsMike) [inline internal/server/iigointernal/executive.go:184]: It doesn't seem to return an error here. Is this intentional?

File: internal/server/iigointernal/executive_test.go [hunk 2]
```diff
@@ -472,3 +484,84 @@ func TestGetAllocationRequests(t *testing.T) {
 		})
 	}
 }
+
+func TestPresidentIncureServiceCharge(t *testing.T) {
```
Comment (MikeIsMike) [inline internal/server/iigointernal/executive_test.go]: Incur?

File: internal/server/iigointernal/executive.go (hunk 1 above)
Comment (tobhil98) [reply internal/server/iigointernal/executive.go:184]: Think it should be fine. Basically, we return if it was successful or not. Thanks for checking

**Comment by tobhil98** [comment]
Branch to be merged with the president features

**Pull request closed without merging**